diff --git a/arch/arm/boot/dts/stm32mp157f-dk2.dts b/arch/arm/boot/dts/stm32mp157f-dk2.dts
index d854ed0..bc7ad2a 100644
--- a/arch/arm/boot/dts/stm32mp157f-dk2.dts
+++ b/arch/arm/boot/dts/stm32mp157f-dk2.dts
@@ -94,6 +94,57 @@
 	};
 };
 
+&i2c5 {
+	status = "okay";
+
+	st_ism330dhcx@6b {
+		/* X-STM32MP-MSP01 */
+		compatible = "st,ism330dhcx";
+		reg =<0x6b>;
+		st,int-pin = <2>;
+
+		/* uses GPIO PB12 on X-STM32MP-MSP01 */
+		interrupt-parent = <&gpiob>;
+		/*interrupts = <12 IRQ_TYPE_LEVEL_HIGH>;*/
+		interrupts = <12 IRQ_TYPE_EDGE_RISING>;
+		/* Added below line */
+		st,drdy-int-pin = <1>;
+		status = "okay";
+
+	};
+	iis2dlpc@19 {
+		/*compatible = "st,lis2dw12";*/
+		compatible = "st,iis2dlpc";
+		reg = <0x19>;
+		interrupt-parent = <&gpiof>;
+		interrupts = <0 IRQ_TYPE_EDGE_RISING>;
+	};
+	mag40@1e {
+		compatible = "st,lsm303ah_magn";
+		reg = <0x1e>;
+		interrupt-parent = <&gpioa>;
+		interrupts = <8 IRQ_TYPE_EDGE_RISING>;
+	};
+};
+
+
+&spi5 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&spi5_pins_a>;
+	pinctrl-1 = <&spi5_sleep_pins_a>;
+	/*status = "disabled";*/
+	cs-gpios = <&gpiof 6 0>;
+	status = "okay";
+
+	spidev@0 {
+		compatible = "semtech,sx1301";
+		spi-max-frequency = <5000000>;
+		reg = <0>;
+	};
+};
+
+
+
 &ltdc {
 	status = "okay";
 
